iib/workers/tasks/build_merge_index_image.py

```diff
@@ -285,3 +287,63 @@ def handle_merge_request(
     set_request_state(
         request_id, 'complete', 'The index image was successfully cleaned and updated.'
     )
+
+
+def filter_invalid_bundles(bundle_paths, valid_ocp_version):
+    """
+    Filter bundles which are invalid for the index ocp_version.
+
+    :param list bundle_paths: list of bundles to be validated against the index ocp version.
+    :param str valid_ocp_version: the index ocp version against which the bundles will be validated.
+    :return: a list of valid bundles which satisfy the index ocp_version
+    :rtype: list
+    :raises IIBError: if bundles have a semantically invalid version set.
+
+           |  "v4.5"   |   "=v4.6"    | "v4.5-v4.7" | "v4.5,v4.6"
+    ---------------------------------------------------------------
+    v4.5   | included  | NOT included |  included   |  included
+    ---------------------------------------------------------------
+    v4.6   | included  |   included   |  included   |  included
+    ---------------------------------------------------------------
+    v4.7   | included  | NOT included |  included   |  included
+    """
+    index_valid_bundles = []
```

Comment:
I think `valid` and `invalid` is not the correct terminology. `index_valid_bundles` contains a list of bundle images that are allowed to be included in the index image, while `invalid_version_bundles` contains a list of bundle images that contain unsupported metadata. There's a third category which are the bundle images that contain the correct metadata, but should not be added to the index image. 

IMO, this function would be better off if it only processed a single bundle at a time, and it only returned a boolean which indicates whether or not the bundle should be added to the index. If the bundle image contains unsupported metadata, log a warning and return False.